### ISSUE TITLE
Setting Unicode caps to Android appium driver throws error for very first run.

### ISSUE THREAD

**Issue by kbalakumar99** [appium/appium#12418]
## The problem

I am running appium and android inside Docker container. The **first test case** running after starting container **fails** if the "unicodeKeyboard" capability is set to the driver. 

Cases running after the first case don't have this issue. As I will be starting the docker container on triggering test, this case is crucial for us. 

## Environment

* Appium version : 1.12.1 / 1.12.0
* Last Appium version that did not exhibit the issue (if applicable): 1.11.1

* Node.js version (unless using Appium.app|exe):  v11.13.0
* Npm or Yarn package manager: 6.7.0
* Mobile platform/version under test: Android 9
* Real device or emulator/simulator: 
  Name: emulator
  Device: Nexus 6 (Google)
  Target: Google APIs (Google Inc.)
  Based on: Android API 28 Tag/ABI: google_apis/x86
*OS: ubuntu:18.10 

## Details

in v1.12.0 this fix is made, which may be the cause of this issue.
> Fix exception caused when Appium Settings fails to provide geolocation #493

tested in v1.11.1 and it is working fine

## Link to Appium logs

https://gist.github.com/kbalakumar99/21e70b38a37be797b7355bf747413ce7#file-log-txt

## Code To Reproduce Issue [ Good To Have ]

Any sample test Code just to start the test is enough to reproduce the issue.

**Comment by mykola-mokhnach** [comment]
> [AndroidDriver] Ignored error while installing Appium Settings helper: 'Error executing adbExec. Original error: 'Command '/home/android/sdk/platform-tools/adb -P 5037 -s emulator-5554 push /home/android/.nvm/versions/node/v11.13.0/lib/node_modules/appium/node_modules/io.appium.settings/apks/settings_apk-debug.apk /data/local/tmp/appium_cache/728413456b86856a2003edeb524d6bd577ca6418.apk' exited with code 1'; Stderr: ''; Code: '1''. Manually uninstalling the application with package id 'io.appium.settings' may help. Expect some Appium features may not work as expected unless this problem is fixed.

I'd say such problem is expected.This must a problem with the emulator that it cannot properly push the apk. Although @KazuCocoa has added a retry for utilities startup into appium@beta

**Comment by kbalakumar99** [comment]
@mykola-mokhnach  Need any other info?
If this is the problem with the emulator, how Can I overcome this? To speed up emulator boot up, I am launching the emulator manually before starting the test and wait for it to complete its boot up using ADB commands.

**Comment by kbalakumar99** [comment]
> > [AndroidDriver] Ignored error while installing Appium Settings helper: 'Error executing adbExec. Original error: 'Command '/home/android/sdk/platform-tools/adb -P 5037 -s emulator-5554 push /home/android/.nvm/versions/node/v11.13.0/lib/node_modules/appium/node_modules/io.appium.settings/apks/settings_apk-debug.apk /data/local/tmp/appium_cache/728413456b86856a2003edeb524d6bd577ca6418.apk' exited with code 1'; Stderr: ''; Code: '1''. Manually uninstalling the application with package id 'io.appium.settings' may help. Expect some Appium features may not work as expected unless this problem is fixed.
> 
> I'd say such problem is expected. This must be a problem with the emulator that cannot properly push the apk. Although @KazuCocoa has added a retry for utilities startup into appium@beta

I forgot to mention another issue.

I tried to solve this problem by installing the settings apk using ADB commands before starting driver.
This caused failure in installation of uiautomator2-server apk. 
And when I installed uiautomator2-server using ADB, pushing the test apk failed. I can give you logs if you want.

**Comment by mykola-mokhnach** [comment]
Most likely your boot monitoring script does not properly sync with the emulator state, so it is still loading some services in the background while push is being performed, which causes it to fail

**Comment by kbalakumar99** [comment]
> Most likely your boot monitoring script does not properly sync with the emulator state, so it is still loading some services in the background while the push is being performed, which makes the whole flow to fail

So, how this works in v1.11.1? 
Is there any better script to monitor booting of the emulator?

**Comment by mykola-mokhnach** [comment]
You can try if appium@beta works for you

**Comment by mykola-mokhnach** [comment]
Try to also get the logcat output. Maybe, it contains some useful data

**Comment by kbalakumar99** [comment]
Okay sure.

**Comment by kbalakumar99** [comment]
It is working in appium@beta

https://gist.github.com/kbalakumar99/303a6d27805bda716a73d078203b9234#file-appium-beta-log

**Comment by mykola-mokhnach** [comment]
Any chance to get the logcat? Maybe, we could improve Simulator state detection if we had this information

**Comment by kbalakumar99** [comment]
Do you need Logcat for the failure case? i.e., for version '1.12.1'

**Comment by mykola-mokhnach** [comment]
Yes, I need the logcat for the settings application deployment failure

**Comment by kbalakumar99** [comment]
https://gist.github.com/kbalakumar99/11597a91f914b502fb91093707ed9b5e#file-logcat-txt

**Comment by mykola-mokhnach** [comment]
Nothing helpful in the logcat output. Lets assume it is working in beta. Let us know if there are still problems like this after the next stable Appium version is released.

**Comment by sharathhatter** [comment]
I am still facing the issue,

appium 1.12.1

An unknown server-side error occurred while processing the command. Original error: Error executing adbExec. Original error: 'Command '/Users/mobility/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell pm install -r /data/local/tmp/appium_cache/0cb99c26c6dc89083fa5781b5c4c077ea179ec8d.apk' exited with code 1'; Stderr: 'Failure [INSTALL_FAILED_UPDATE_INCOMPATIBLE: Package com.test.mobileapp signatures do not match the previously installed version; ignoring!]'; Code: '1'

**Comment by KazuCocoa** [comment]
Have you installed the `com.test.mobileapp` before with another signature?
It is probably fixed after uninstalling `com.test.mobileapp` on the test device by hand.
